tests/test_analyzer.py

```diff
@@ -74,9 +74,13 @@ def test_correlation_helper_for_data_with_three_rows(self):
     def test_correlation_helper_for_data_with_four_rows(self):
         # this should compute marginal correlations and return a unity
         # matrix for partial correlations
-        retval = Analyzer.correlation_helper(self.df_features[:4], 'sc1', 'group')
+        # it should also raise a UserWarning
+        with warnings.catch_warnings(record=True) as w:
+            retval = Analyzer.correlation_helper(self.df_features[:4], 'sc1', 'group')
         assert_equal(retval[0].isnull().values.sum(), 0)
         assert_almost_equal(np.abs(retval[1].values).sum(), 0.9244288637889855)
+        assert issubclass(w[-1].category, UserWarning)
```

Comment:
```suggestion
        assert issubclass(warning_list[-1].category, UserWarning)
```

Is there supposed to be multiple warnings here? If not, I'd suggest using `[0]`.